### ISSUE TITLE
Saying no to the prompt to hit an ally when channeling a spell waits a turn

### ISSUE THREAD

**Issue by WizardIke** [crawl/crawl#3506]
**PLATFORM and OS or BROWSER:**

webtiles

**DCSS VERSION AND GAME INFORMATION:**

Version: 0.31

**DESCRIPTION:**

If you cast flame wave and an ally walks into it while you are channeling it with a god that doesn't like you hurting allies and say no to the prompt, the game waits a turn without channeling instead of just canceling channeling the spell.

**EXTRA FILES:**

![Screenshot 2024-02-05 172547](https://github.com/crawl/crawl/assets/26264063/78e26286-ecff-48f6-ab32-00e23fe99b12)
![Screenshot 2024-02-05 172607](https://github.com/crawl/crawl/assets/26264063/7e6fa30f-cd65-4041-974e-b416019a58b6)

**Comment by projectNoob** [comment]
I agree and in addition the game should provide an option to stop channeling and wait.
